packages/web/src/store/errors/reportToSentry.ts

```diff
@@ -12,28 +12,52 @@ const Levels: { [level in ErrorLevel]: SeverityLevel } = {
   Log: 'log'
 }
 
+type ConsoleLoggingMethod = keyof Pick<
```

Comment:
@rickyrombo couple of my reportToSentry changes, main things I did:
- make console method match the requested level (console.error/.log/etc)
- reduced it down to one console log instead of one basic one + a second one with "additional info"
- allowed passing a string instead of an error object
- added tags as an arg

I know you have opinions against #2 so we can discuss it here. Also I was wondering, if I `captureException` and `console.error` the same error will it show up twice? It doesnt seem to be doing it when I trigger logs locally but I'm not convinced

Reply:
I see your point about not allowing strings so that people aren't allowed to not pass along the original error object. 
My POV is that I think it should be easier to customize the message so that we can start grouping errors better and not rely on runtime generated messages. I know sentry can somewhat group things but I think we've seen that norm
So like with an error reading like: `TypeError: Cannot read properties of null (reading 'file')'` 
which is created by doing
```tsx
catch(error) {
  reportToSentry({ error, ...etc})
```
its instead 
```tsx
catch(error) {
  reportToSentry({ error:"Error uploading file or whatever", additionalInfo: { error },  ...etc})
```

Reply:
> I know you have opinions against (2) so we can discuss it here.

There's a few places in the code that get the error message out of the error before pushing that error to sentry, which means we lose the stack and the juicy goodness that can flow through the error. I want to make it hard to accidentally do this - even making the error locally when calling `reportToSentry` will have more context and be more interesting I think than just sending the string iiuc. Biggest offender here is the `handleError` action, which is doing its best to be a serializable plain object but in doing so loses all the error context goodies. I don't feel super strongly here though, we should have other folks weigh in!

> Also I was wondering, if I captureException and console.error the same error will it show up twice? It doesnt seem to be doing it when I trigger logs locally but I'm not convinced

My understanding is that it would, yes, but could be wrong?

Reply:
Just saw your second post - yeah I think that's the right idea, but I think we lose _what_ error happened while uploading a file, and the bucket might be _too_ broad - when we resolve that issue, we'll resolve _all_ errors while uploading which isn't what we want necessarily. It's a tough balance to strike but I think having at least the error message for grouping is good - we just have to make sure to not clutter the error message with specific info and leave that to the "additionalInfo" or context

Reply:
Yeah I think preserving the stack is a good point. I thought about it and I think the main thing I care about is adding extra context to the error message, but to your point below, I can do that via `name` as well so I think that's what Ill do for now.
I have a feeling we'll still end up with error messages that are hard to group but I think we can hopefully start addressing them case by case. It can also be nice if the issues are grouped by the message too.

Reply:
some updates for posterity; 
Marcus and I tested several things and found that Sentry automatically does some deduping of error logging, so doing a `console.error` alongside a `captureException` should be fine which is exciting.
Also, I removed the "pass a string along" feature & we decided to update `name` to "append" the existing name.

For now this is looking good, although in the future we're thinking about changes that encourage people to construct and populate their own scope instead of just waiting to log data in the catch fn